scijava-ops-legacy/src/test/java/org/scijava/legacy/types/DatasetTypeExtractorTest.java

```diff
@@ -65,6 +67,11 @@ public void testDatasetTypeExtractor() {
 		var actual = new DatasetTypeExtractor().reify(reifier, ds);
 		var expected = Types.parameterize(ImgPlus.class, new Type[] { new Any() });
 		Assertions.assertEquals(expected, actual);
+		OpEnvironment env = OpEnvironment.build();
```

Comment:
Ahh, there is an `NB` comment above that is actually obsolete, isn't it?

Reply:
Heh, it's complicated, but it still holds. JPMS shenanigans.

The `scijava-ops-legacy` module does not `use` `TypeExtractor`, so no TypeExtractors are found using the Discoverers in the declaration of the `DefaultTypeReifier`. Therefore, when we *directly* call the `DatasetTypeExtractor`, the recursive call to reify the `ImgPlus` finds no extractors and returns `ImgPlus<Any>`.

In the line you linked, the `TypeReifier` is constructed within the SciJava Ops Engine, whose module *does* `use` the `TypeExtractor` implementations, so it can find all of the `TypeExtractor` implementations, and we get back the correct type. Note that `scijava-ops-legacy` does *not* depend on scijava-ops-image, so the reifier actually being used here is the very slow `IterableTypeExtractor`.

I described the situation in 449f98f1